orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/MonitoredDeployStrategy.groovy

```diff
@@ -0,0 +1,354 @@
+        ]
+
+        log.info("Adding `Disable $p% of Desired Size` stage with context $disableContext [executionId=${stage.execution.id}]")
+
+        def disablePortionStage = newStage(
+          stage.execution,
+          disableServerGroupStage.type,
+          "Disable $p% of Traffic on ${source.serverGroupName}",
+          disableContext,
+          stage,
+          SyntheticStageOwner.STAGE_AFTER
+        )
+        stages << disablePortionStage
+      }
+
+      if (mdsd.deploymentMonitor.id) {
+        evalContext.currentProgress = p
+
+        stages << newStage(
+          stage.execution,
+          evaluateDeploymentHealthStage.type,
+          "Evaluate health of deployed instances",
+          evalContext,
+          stage,
+          SyntheticStageOwner.STAGE_AFTER
+        )
+      }
+    })
+
+    // only scale down if we have a source server group to scale down
+    if (source && stageData.scaleDown) {
+      if (stageData?.getDelayBeforeScaleDown()) {
+        def waitContext = [waitTime: stageData?.getDelayBeforeScaleDown()]
+        stages << newStage(
+          stage.execution,
+          waitStage.type,
+          "Wait Before Scale Down",
+          waitContext,
+          stage,
+          SyntheticStageOwner.STAGE_AFTER
+        )
+      }
+
+      def scaleDown = baseContext + [
+        allowScaleDownActive         : false,
+        remainingFullSizeServerGroups: 1,
+        preferLargerOverNewer        : false
+      ]
+      stages << newStage(
+        stage.execution,
+        scaleDownClusterStage.type,
+        "scaleDown",
+        scaleDown,
+        stage,
+        SyntheticStageOwner.STAGE_AFTER
+      )
+    }
+
+    // TODO(mvulfson): Unpin
+    if (source) {
+
+    }
+
+    // TODO(mvulfson): Shrink cluster
+
+
+    if (mdsd.deploymentMonitor.id) {
+      stages << newStage(
+        stage.execution,
+        broadcastDeployCompletedStage.type,
+        "Broadcast Monitored Deployment Completed",
+        evalContext,
+        stage,
+        SyntheticStageOwner.STAGE_AFTER
+      )
+    }
+
+    return stages
+  }
+}
+
+// TODO(mvulfson): Move this someplace nice
+class Capacity {
+  int min
+  int max
+  int desired
+}
+
+class FailureActions {
+  enum RollbackType {
+    None,
+    Automatic,
+    Manual
+  }
+
+  RollbackType rollback
+  boolean destroyInstances
+}
+
+class DeploymentMonitor {
+  String id
+  Map<String, Object> parameters
+}
+
+class MonitoredDeployStageData extends StageData {
+  List<Integer> deploySteps = []
+  Capacity targetCapacity
+  int maxRemainingAsgs
+  int scaleDownOldAsgs
+  FailureActions failureActions
+  DeploymentMonitor deploymentMonitor
+  int deployMonitorHttpRetryCount
+
+  Map getChildStageContext() {
+    def context = [
+            deploymentMonitor: deploymentMonitor,
+    ]
+
+    return context
+  }
+
+  @JsonProperty("deploy.server.groups")
+  Map<String, Set<String>> deployServerGroups = [:]
+
+  //Capacity originalCapacity
```

Comment:
why commented out?

Reply:
hm.... good question. I had a great idea for it, and then I forgot so i commented it out to keep thinking about it... :(